### ISSUE TITLE
Speed up unpacking software on Windows

### ISSUE THREAD

**Issue by snooppr** [pyinstaller/pyinstaller#7473]
I collect the whole project into one file (numpy and scipy are present). All is well, the compiled project has a size of 60MB. But when you open the file, the launch takes 19 seconds - it's very, very slow.
IO unpacking speed on the monitor ~ 6Mb/s.

While the project was running, I looked into the unpacked project folder: C:\Users\User\AppData\Local\Temp "_MEI47602". Its total size is 123MB, the number of files is 260.
I made a rar archive "_MEI47602" not with strong compression. The archive "_MEI47602.rar"= 40Mb after compression weighs less than one file compiled in pyinstaller (60Mb).

I am unpacking "_MEI47602.rar" and it takes attention! jus**t 4 seconds vs pyinstaller 19 seconds!**

OS used Windows 7 32bit, python3.8 32bit, pyinstaller v 5.7.0.

Please do something to speed up the loader or add custom project compression options to a single file (eg less compression but faster decompression). It seems that this problem only affects Windows, on GNU/Linux the same thing works much faster.
ps/ in 2023 for Linux, you generally need to unpack the project into RAM in a split second, and not on the HDD.

**Comment by bwoodsend** [comment]
If Windows is taking that long then it's your antivirus causing the congestion by scanning each file as it's unpacked. Deflate compression is designed to be as fast as possible to decompress. On my machine, it takes under a second to start.

**Comment by rokm** [comment]
I cannot reproduce this, either. Although we have seen reports of slow downs (even more drastic ones) in #6855.

FWIW, a program importing `numpy` and `scipy`,
<details><summary>progam.py</summary><p>

```python
print("Started program!")
import numpy
import scipy.cluster
import scipy.constants
import scipy.datasets
import scipy.fft
import scipy.fftpack
import scipy.integrate
import scipy.interpolate
import scipy.io
import scipy.linalg
import scipy.misc
import scipy.ndimage
import scipy.odr
import scipy.optimize
import scipy.signal
import scipy.sparse
import scipy.spatial
import scipy.special
import scipy.stats
print("Hello world!")
```

</p></details>

comes up at around 146 MB in onedir mode and 51 MB in onefile mode in my test environment. 

Extraction of onefile archive in my 64-bit Windows 10 VM takes a bit more than a second, as measured by [instrumented bootloader](https://github.com/pyinstaller/pyinstaller/commit/54005f51dcd013b9be5bff6e7842dfceff1d14ec):
```
>dist\program.exe
PERF: extraction start at 17229967...
PERF: extraction end at 17231233...
PERF: extraction took 1266 ms!
Started program!
Hello world!
```

On the host linux system, it's well under a second:
```
$ ./dist/program
PERF: extraction start at 53202071...
PERF: extraction end at 53202678...
PERF: extraction took 607 ms!
Started program!
Hello world!
```

---

So unless you can figure out what exactly is slowing down extraction on your system (assuming it is not an AV), there's not much we can do about it.

**Comment by snooppr** [comment]
@bwoodsend  There is no antivirus on the OS. 

I have some ideas for tests, it is related to packaging under Windows 7. But the Pyinstaller documentation says that Windows 7 is not supported.  The assembly of this one file takes just as long to open on a powerful Windows 10 PC (On another PC, also with windows 7 maximum, unpacking this project takes > 10 seconds.  Wouldn't that be of interest?).

It seems strange to me: to close the problem, which is not mentioned for the first time, and hang up the label "invalid". Put all the blame on the antivirus, which is not even installed

A solution, step by step, could well be found, but if it is not so interesting "label invalid", the solution will not be published.: a bug in pyinstaller.

upd: 
This problem only affects "onefile", in folder mode this project starts instantly.  In the onefile mode, the problem is with unpacking, as I wrote above, the extraction speed is 6Mb / s.

**Comment by bwoodsend** [comment]
>  There is no antivirus on the OS.

Even if you didn't install one, there is always Windows Defender.

> It seems strange to me: to close the problem, which is not mentioned for the first time, and hang up the label "invalid". Put all the blame on the antivirus, which is not even installed
>
> A solution, step by step, could well be found, but if it is not so interesting "label invalid", the solution will not be published.: a bug in pyinstaller.

What do you reasonably expect us to do if we can't reproduce it to investigate it ourselves, CI/CD can't reproduce it and you won't investigate it.

**Comment by snooppr** [comment]
>  Even if you didn't install one, there is always Windows Defender.

It is also disabled at the service level.


After digging around, I found what was wrong.  The results shocked me.

On different OS Windows 7 (one, 32bit, another 64bit).  The compiled project (built on Windows 32bit) runs in 19 seconds on both OSes.  What I did, I uninstalled python3.7 on Windows 64bit, ran the project, no change, the decompression speed is the same: 6Mb / s launch 19sec.  The project works, but it takes a long time to start.  Attention!  I installed on Windows/64 python3.8/64 without dependencies, bare.  And just launched the old compiled project, it started up in 5 seconds!.  I uninstalled python3.8/64bit again and ran the old compiled project again, decompress time 19 seconds.  I restarted the PC, but the project starts in 19 seconds.  I installed python3.8 again, but the 32bit versions.  Ran the project and it runs again within 5 seconds.  CONCLUSION: how can the python installed in the OS affect the launch of the project?!  It is fantastic.  The compiled project does not use third-party python in any way (I check in the project, I see that python is compressed into one file).



UPD.
I compiled about 30 copies of the project for the sake of testing, narrowing down the search for the problem.  I can unequivocally state that the presence of python installed on the system somehow affects the project start time, cutting the call in half (on my hardware).  Fact.  I've also found that the slowdown only triggers code that needs scipy and numpy.  And I also tested the project on another hardware, but also with Windows 7/64, there was never python there and the code runs after 8 seconds.  I also checked the project on GNU/Linux, it runs within 1 second.
